CMakeLists.txt

```diff
@@ -306,6 +314,15 @@ else()
 	endif (MSVC)
 endif()
 
+if (NOT MSVC)
+	if (abi STREQUAL "c++03")
+		add_definitions(-std=${abi})
+	elseif (abi STREQUAL "c++11" OR abi STREQUAL "c++14")
+		# Also silence auto_ptr because it throws lots of warnings.
+		add_definitions(-std=${abi} -Wno-deprecated-declarations)
```

Comment:
those warnings are supposed to be suppressed. at least for msvc, gcc and clang. is that broken? or do you use a different compiler?